src/odemis/driver/actuator.py

```diff
@@ -3003,7 +3003,7 @@ def _doReference(self, axes, omit_referenced):
                 # Skip axes which are already referenced or cannot be referenced
                 stage_axes = {ax for ax in stage_axes if not self.stage.referenced.value.get(ax, True)}
             logging.debug("Referencing stage axes %s.", stage_axes)
-            f = self.stage.reference(stage_axes)
+            f = self.stage.reference(set(stage_axes))
```

Comment:
Can you add a comment why you need to do this please?

Reply:
reference() expects a set. So it sort of makes sense... but I'd suggest that a clearer way is to directly convert stage_axes when it's read: line 3001 becomes `stage_axes = set(self.stage.axes.keys())`.

Reply:
Does that then also work with line 3004, or should that then also be updated to 

`stage_axes = set(ax for ax in stage_axes if not self.stage.referenced.value.get(ax, True))`

Reply:
Line 3004 returns a set already.
In Python, it's a little confusing, but `{1, 2}` is a set, and `{1: "a", 2: "b"}` is a dict.
So `{a for a in....}` is a set, and `{a: b for a, b in ....}` is a dict.